pkg/binding/binding-controller.go

```diff
@@ -575,14 +577,14 @@ func (c *Controller) setupBindingInformer(ctx context.Context) error {
 		AddFunc: func(obj interface{}) {
 			bdg := obj.(*v1alpha1.Binding)
 			logger.V(5).Info("Enqueuing reference to Binding because of informer add event", "name", bdg.Name, "resourceVersion", bdg.ResourceVersion)
-			c.workqueue.Add(bindingPolicyRef(bdg.Name))
+			c.workqueue.Add(bindingRef(bdg.Name))
 		},
 		UpdateFunc: func(old, new interface{}) {
 			oldBdg := old.(*v1alpha1.Binding)
 			newBdg := new.(*v1alpha1.Binding)
-			if oldBdg.Generation != newBdg.Generation {
+			if oldBdg.ResourceVersion != newBdg.ResourceVersion {
```

Comment:
Some notes on this change: I'm not sure why we use Generation here. What I observed is that (status controller) updating status of bindings won't increase Generation, thus would inform the binding controller on the updates of the Condition.

Use ResourceVersion may not be a good idea as well. Maybe we just remove this if condition?

Reply:
In general the filtering here should be only sensitive to stuff that matters. For example, changes to labels and annotations do not matter. In this particular case, I think the right thing to do is compare Generation (as a summary of Spec) and also compare Status.Conditions and Status.Errors (it would suffice to do an element-by-element comparison, without trying to recognize that reordering is insignificant).

Reply:
Done.